src/frontend/hooks/server/projects.ts

```diff
@@ -28,19 +19,29 @@ export function useGetOwnRole() {
   return useOwnRoleInProject({projectId});
 }
 
-export function useGetRemoteArchives() {
-  const {projectId} = useActiveProject();
-  const {data: members, error, isRefetching} = useManyMembers({projectId});
+// TODO: Ideally this is handled in @comapeo/core (https://github.com/digidem/comapeo-core/issues/1031)
+export type ArchiveServerMemberInfo = MemberInfo & {
+  deviceType: 'selfHostedServer';
+  selfHostedServerDetails: NonNullable<MemberInfo['selfHostedServerDetails']>;
+};
 
-  const archives = useMemo(() => {
-    return members?.filter(m => m.deviceType === 'selfHostedServer') ?? [];
-  }, [members]);
+// TODO: Ideally this is handled in @comapeo/core (https://github.com/digidem/comapeo-core/issues/1031)
+function isActiveArchiveServerMember(
+  member: MemberInfo,
+): member is ArchiveServerMemberInfo {
+  if (member.deviceType !== 'selfHostedServer') return false;
+  if (!member.selfHostedServerDetails) return false;
+  if (member.role.roleId !== MEMBER_ROLE_ID) return false;
+  return true;
+}
 
-  return {
-    data: archives,
-    error,
-    isRefetching,
-  };
+export function useActiveArchiveServer({
+  projectId,
+}: {
+  projectId: string;
+}): ArchiveServerMemberInfo | undefined {
+  const {data: members} = useManyMembers({projectId});
+  return members.find(isActiveArchiveServerMember);
```

Comment:
I think this is right, but I just want to make sure. Each project should only be connected to one archive server. I know that we technically can be part of many archive servers, but we don't allow that on the client side right now, correct?

Reply:
Yeah that's how I understand it. Core allows you to connect to multiple archive servers, but in the client side we only support connecting to a single one.